src/Facebook/FacebookBatchRequest.php

```diff
@@ -189,8 +191,19 @@ public function convertRequestsToJson()
     {
         $requests = [];
         foreach ($this->requests as $request) {
-            $attachedFiles = isset($request['attached_files']) ? $request['attached_files'] : null;
-            $requests[] = $this->requestEntityToBatchArray($request['request'], $request['name'], $attachedFiles);
+            $options = [];
+
+            if (!empty($request['name'])) {
```

Comment:
To solve the issues, check here for `null !== $request['name']`, and in the `add` function, cast the `$key` to a string :)